invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,73 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        :param identity: The user identity performing the operation.
+        :param data: A list of tuples containing the record ID and record data.
```

Comment:
just for consistency reasons: are we using tuples also in the member bulk operations? it would be good to have a uniform data structure for all the bulk operations in the application

Reply:
In members seems that we simply return True after processing everything, see here: https://github.com/inveniosoftware/invenio-communities/blob/master/invenio_communities/members/services/service.py#L232

I got inspired here with what we did for the including multiples records in a community, where we return a tuple of processed records and the errors. 

In this case we changed it a bit because ideally we want to keep track of the error, to also have the record data and operation tupe that caused the error.

Reply:
I see that the result type for the new service call is a big discussion topic in this PR.

Should we create a new `BulkResult` class, with an appropriate `BulkResultItem` (or better named?) container type to wrap all the necessary fields/info in an organized way? The main issue I see is that we have two ways to represent errors from service calls in RDM:

- Populating the `errors` list and including it in the result (e.g. for schema loading validation errors, or errors in service components)
  - We're very inconsistent about what goes in this list though, sometimes it's a [Marshmallow-style dictionary](https://github.com/inveniosoftware/invenio-drafts-resources/blob/607597db3bcaff600462a05521d3635c9032d481/invenio_drafts_resources/services/records/components/base.py#L114-L121), and other times just [a string](https://github.com/slint/invenio-rdm-records/blob/0a14ba9dc06645420d7543275b56a01dc88ede07/invenio_rdm_records/services/components/access.py#L38-L40)
- Raising an exception so that the caller can handle it
  - I would prefer being more "transactional" with the result types, meaning that raised exceptions are just wrapped, and it's up to the caller to handle them depending on their context
  - But this also can be dangerous since if the caller doesn't check if the results have errors, they might miss it (while a raised exception "forces" you to handle it)

My preference for bulk operations would be not to raise but instead wrap everything.

Reply:
I agree with the BulkResult solution and BulkResultItem potentially (isn't it just ResultItem? I think only the "collection" needs a new class)

Reply:
PR updated with the above mentioned changes